code/game/objects/items/stacks/sheets/leather.dm

```diff
@@ -243,23 +244,50 @@
 		..()
 
 
-//Step two - washing..... it's actually in washing machine code.
+//Step two - washing..... it's actually in washing machine code AND in sink code!
 
 //Step three - drying
 /obj/item/stack/sheet/wetleather/process()
+	dryingspeeddesc = null
 	var/turf/location = get_turf(src)
 	if(!location)
 		return
+	var/totaldrying = 0
 	var/datum/gas_mixture/environment = location.return_air()
-	if(environment.temperature >= drying_threshold_temperature)
-		wetness--
-		if(wetness <= 0)
-			if(amount)
-				var/obj/item/stack/sheet/leather/L = drop_stack(/obj/item/stack/sheet/leather, loc, amount)
-				L.source_string = source_string
-				L.name = source_string ? "[source_string] leather": "leather"
-				use(amount)
+	if(environment.temperature >= drying_threshold_temperature + 21) //Original pre-buff temperature behavior, default 1 minute
+		totaldrying = totaldrying + 6
+	else if(environment.temperature >= drying_threshold_temperature) //(Default variable value) Room temperature drying, default 6 minutes
+		totaldrying = totaldrying + 1
+	wetness = wetness - totaldrying
+	if(wetness <= 0)
+		if(amount)
+			var/obj/item/stack/sheet/leather/L = drop_stack(/obj/item/stack/sheet/leather, location, amount)
+			L.source_string = source_string
+			L.name = source_string ? "[source_string] leather": "leather"
+			use(amount)
+		return
+	switch(totaldrying)
+		if(1 to 3)
+			dryingspeeddesc = "drying slowly."
+		if(4 to INFINITY)
+			dryingspeeddesc = "drying quickly!"
+		else
+			dryingspeeddesc = "not drying."
 
+/obj/item/stack/sheet/wetleather/examine(mob/user)
+	..()
+	switch(round(1-(wetness/initial(wetness)),0.01)*100) //returns dryness percent
+		if(0 to 20)
+			to_chat(user, "<span class='info'>It's soaking wet!</span>")
+		if(20 to 40)
+			to_chat(user, "<span class='info'>It's wet.</span>")
+		if(40 to 60)
+			to_chat(user, "<span class='info'>It's damp.</span>")
+		if(60 to 80)
+			to_chat(user, "<span class='info'>It's still a little damp in places.</span>")
+		if(80 to 100)
+			to_chat(user, "<span class='info'>It's almost completely dry!</span>")
+	to_chat(user, "<span class='info'>It's [dryingspeeddesc]</span>")
```

Comment:
You have overlapping ranges here (ie 0 to 20 and 20 to 40). Not sure if this will runtime if you examine something at 20 wetness or just use the lower value, but it'd probably be better to remove the overlaps entirely.

Reply:
Gonna fix it. The reference says that the blocks are parsed in order, so overlaps in theory don't matter... but there's a warning that overlap behavior is undefined. Very cursed. Won't let that happen on my watch! Can't have undefined behavior! It could do anything! It could even spawn a boat!